nsxt/provider.go

```diff
@@ -541,6 +556,33 @@ func (processor remoteBasicAuthHeaderProcessor) Process(req *http.Request) error
 	return nil
 }
 
+func applyLicense(c *nsxt.APIClient, licenseKey string) error {
+	if c == nil {
+		return fmt.Errorf("API client not configured")
+	}
+
+	license := licensing.License{LicenseKey: licenseKey}
+	_, resp, err := c.LicensingApi.CreateLicense(c.Context, license)
+	if err != nil {
+		return fmt.Errorf("Error during license create: %v", err)
+	}
+	if resp.StatusCode != http.StatusOK {
+		return fmt.Errorf("Unexpected status returned during license create: %v", resp.StatusCode)
+	}
+
+	return nil
+}
+
+func configureLicenses(d *schema.ResourceData, clients *nsxtClients) error {
+	for _, lic_key := range d.Get("license_keys").([]interface{}) {
```

Comment:
I suggest we only do this is d.HasChange("license_keys").
Also shouldn't we handle license removal as well?
d.GetChange can help determine which ones were added and which ones removed

Reply:
It seems as if provider attributes are not persisted, and d.HasChange returns true even when there has not been a change in configuration. d.GetChange always produces an empty list and the current list instead of the previous configuration in this case.  It may make sense for License to be a resource if removal is necessary. Should we keep license in the provider and not handle removal, or make it a resource and document the need for explicit depends_on?

Reply:
I see.. Removing a license is probably a rare use case.. I'm ok with leaving this as is (it would be nice to leave a comment about this in the code and in docs).